[P0] Fix CI pipeline - sbt command not found

## Summary

The GitHub Actions CI pipeline fails immediately because `sbt` is not installed on the runner.

## Error

```
/home/runner/work/_temp/36b25a6e-6924-4efa-bfe4-149d00379254.sh: line 1: sbt: command not found
Error: Process completed with exit code 127.
```

## Root Cause

The `setup-java` action with `cache: 'sbt'` only caches sbt dependencies - it does **not** install sbt itself. The runner has no sbt binary available.

## Fix

Add the official sbt setup action before running sbt commands:

```yaml
- name: Set up sbt
  uses: sbt/setup-sbt@v1
```

## Updated Workflow

```yaml
steps:
  - name: Checkout code
    uses: actions/checkout@v4

  - name: Set up JDK 17
    uses: actions/setup-java@v4
    with:
      distribution: 'temurin'
      java-version: '17'

  - name: Set up sbt
    uses: sbt/setup-sbt@v1

  - name: Compile
    run: sbt compile

  - name: Run tests
    run: sbt test
```

## Priority

**P0 (Critical)** - CI is completely broken, no tests can run.

## Affected Workflow

- `.github/workflows/ci.yml`
- Run #1: https://github.com/VledicFranco/constellation-engine/actions/runs/1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -22,6 +22,9 @@ jobs:
           java-version: '17'
           cache: 'sbt'
 
+      - name: Set up sbt
+        uses: sbt/setup-sbt@v1
+
       - name: Compile
         run: sbt compile
 

